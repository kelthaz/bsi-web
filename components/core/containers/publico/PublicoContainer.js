import Footer from '../../footer/Footer';
import Header from '../../header/Header';

const PublicoContainer = ({ pageComponent, servicesData }) => {
  const { component: Component } = pageComponent;

  return (
    <>
      <Header />
      <Component {...servicesData} />
      <Footer />
    </>
  );
};

export default PublicoContainer;
