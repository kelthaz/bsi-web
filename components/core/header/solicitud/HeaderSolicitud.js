import React from 'react';

const HeaderSolicitud = () => {
  return <div></div>;
};

export default HeaderSolicitud;
