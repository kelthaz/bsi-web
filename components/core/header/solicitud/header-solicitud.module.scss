@import '../../../../styles/scss/modules/var';
@import '../../../../styles/scss/modules/mixins';

.container-header {
  align-items: center;
  border-bottom: 1px solid $blue-shadow;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px 32px;

  @include media('sm') {
    padding: 14px 16px;
  }
}
