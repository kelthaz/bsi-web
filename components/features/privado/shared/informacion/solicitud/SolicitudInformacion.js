import React from 'react';
import PropTypes from 'prop-types';
import { MORAL } from '../../../../../../constants/persona';

const SolicitudInformacion = ({ informacionSolicitud, tipoPersona }) => {
  const labels = {
    usoCredito: 'Destino del crédito',
    descripcionCredito: 'Razón social',
    domicilioFiscal: 'Domicilio fiscal',
    domicilioComercial: 'Domicilio comercial',
    nombreCompletoRecibe: 'Nombre de quien podría recibir token',
    celularRecibe: 'Teléfono de quien podría recibir token',
    telefonoEmpresa: 'Teléfono del negocio',
    curp: tipoPersona === MORAL ? 'CURP' : 'CURP del representante legal',
    estadoCivil: 'Estado civil del solicitante',
    nombreConyuge: 'Nombre de la Pareja',
    numeroEmpleados: 'Número de empleados',

    montoSolicitado: 'Crédito solicitado',
    montoAprobado: 'Crédito aprobado',

    tasaOrdinaria: 'Tasa ordinaria',
    tasaMoratoria: 'Tasa moratoria',
    comisionApertura: 'Comisión por apertura',
    plazo: 'Plazo del crédito',
    periodicidad: 'Esquema de pago',
    pago: 'Pagos',
    cat: 'CAT',
    tieneCuentaBancoppel: 'Tiene cuenta BanCoppel',
    clabe: 'Número de cuenta BanCoppel',
  };

  if (tipoPersona === MORAL) {
    delete labels.estadoCivil;
    delete labels.nombreConyuge;
  }

  const itemsPerfil = Object.keys(labels);

  return (
    <>
      {itemsPerfil.map((item) => (
        <div key={item} className="row pb-3">
          <div className="col-6">
            <div className="body2">{labels[item]}</div>
          </div>
          <div className="col-6">
            <div className="color-blue-storm body2">{informacionSolicitud[item]}</div>
          </div>
        </div>
      ))}
    </>
  );
};

SolicitudInformacion.propTypes = {
  informacionSolicitud: PropTypes.object.isRequired,
  tipoPersona: PropTypes.string.isRequired,
};

export default SolicitudInformacion;
