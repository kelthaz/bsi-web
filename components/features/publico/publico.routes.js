import {
  AVISO_ROUTE,
  AYUDA_ROUTE,
  BENEFICIOS_ROUTE,
  CREDITO_ROUTE,
  INICIO_ROUTE,
  REQUISITOS_ROUTE,
  SIMULADOR_ROUTE,
} from '../../../constants/routes/publico/publico';
import SelectorSimulador from './simulador/Simulador';
import Ayuda from './ayuda/Ayuda';
import Credito from './credito/Credito';
import Inicio from './inicio/Inicio';
import Requisitos from './requisitos/Requisitos';
import SimuladorRepositorio from '../../../services/simulador/simulador.repositorio';
import AccordionRepositorio from '../../../services/simulador/acordeon.repositorio';
import Beneficios from './beneficios/Beneficios';
import AvisoPrivacidad from './aviso/AvisoPrivacidad';
import AvisoRepositorio from '../../../services/publico/aviso.repositorio';

const publicoRoutes = [
  {
    route: INICIO_ROUTE,
    component: Inicio,
    services: [{ name: 'catalogo', service: SimuladorRepositorio.getSimuladorCatalogo, params: '' }],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'inicio',
    },
  },
  {
    route: CREDITO_ROUTE,
    component: Credito,
    services: [{ name: 'accordionItems', service: AccordionRepositorio.getAccordionPorSector, params: 'credito-pyme' }],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'credito',
    },
  },
  {
    route: REQUISITOS_ROUTE,
    component: Requisitos,
    services: [{ name: 'accordionItems', service: AccordionRepositorio.getAccordionPorSector, params: 'requisitos' }],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'requisitos',
    },
  },
  {
    route: SIMULADOR_ROUTE,
    component: SelectorSimulador,
    services: [{ name: 'catalogo', service: SimuladorRepositorio.getSimuladorCatalogo, params: '' }],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'simulador',
    },
  },
  {
    route: BENEFICIOS_ROUTE,
    component: Beneficios,
    services: [],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'beneficios',
    },
  },
  {
    route: AYUDA_ROUTE,
    component: Ayuda,
    services: [{ name: 'accordionItems', service: AccordionRepositorio.getAccordionPorSector, params: 'ayuda' }],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'ayuda',
    },
  },
  {
    route: AVISO_ROUTE,
    component: AvisoPrivacidad,
    services: [{ name: 'avisoItems', service: AvisoRepositorio.getAviso, params: '' }],
    roles: [],
    feature: 'publico',
    data: {
      tab: 'ayuda',
    },
  },
].map((route) => ({ ...route, feature: 'publico' }));

export default publicoRoutes;
