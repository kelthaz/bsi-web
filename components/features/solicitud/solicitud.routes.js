import cargaDocumentosRoutes from './carga-documentos/carga-documentos.routes';
import datosEmpresaRoutes from './datos-empresa/datos-empresa.routes';
import datosPersonalesRoutes from './datos-personales/datos-personales.routes';
import documentacionRoutes from './documentacion/documentacion.routes';
import ofertaRoutes from './oferta/oferta.routes';

const solicitudRoutes = [
  ...datosPersonalesRoutes,
  // ...datosEmpresaRoutes,
  // ...documentacionRoutes,
  // ...cargaDocumentosRoutes,
  // ...ofertaRoutes
].map((route) => ({ ...route, feature: 'solicitud' }));

export default solicitudRoutes;
