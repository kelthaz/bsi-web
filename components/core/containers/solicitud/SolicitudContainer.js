import { useRouter } from 'next/router';
import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { DATO_PERSONA } from '../../../../constants/formularios';
import { SIMULADOR_ROUTE } from '../../../../constants/routes/publico/publico';
import solicitudRoutes from '../../../features/solicitud/solicitud.routes';
import Step from '../../../shared/step/Step';
import TabInformativo from '../../../shared/tab-informativo/TabInformativo';
import SvgPatronesSolicitud from '../../../svgs/SvgPatronesSolicitud';
import HeaderSolicitud from '../../header/solicitud/HeaderSolicitud';
import ModalActualizar from '../../modals/solicitud/modal-actualizar/ModalActualizar';

const SolicitudContainer = ({ pageComponent, servicesData }) => {
  const { component, data } = pageComponent;
  const [componentPFAE, componentPM] = component;
  const { formulario, paso, step, tab, tipoPersona } = data;
  const { sincronizado, currentStep, tipoPersonaRedux } = useSelector((state) => state.solicitud);
  const { replace } = useRouter();

  let Component = <></>;

  if (tipoPersona === '' && componentPM) {
    Component = tipoPersonaRedux === 'MORAL' ? componentPM : componentPFAE;
  } else {
    Component = componentPFAE;
  }

  const componentsFormulario = solicitudRoutes
    .filter(({ data: dataComponent }) => dataComponent.formulario === formulario)
    .map(({ data: dataComponent, route }) => ({
      step: dataComponent.step,
      paso: dataComponent.paso,
      tipoPersona: dataComponent.tipoPersona,
      action: route,
    }));

  const steps = componentsFormulario.filter(
    (dataComponent) =>
      Number.isInteger(dataComponent.step) &&
      (dataComponent.tipoPersona === '' || dataComponent.tipoPersona === tipoPersonaRedux)
  );

  const tabs = [
    { path: 'datos-personales', label: 'Datos personales' },
    { path: 'datos-empresa', label: 'Datos de tu empresa' },
    { path: 'oferta', label: 'Oferta' },
    { path: 'documentacion', label: 'DocumentaciÃ³n' },
  ];

  useEffect(() => {
    if (!sincronizado && formulario === DATO_PERSONA) {
      replace(SIMULADOR_ROUTE);
    }
  }, []);

  return (
    <>
      <HeaderSolicitud />
      <ModalActualizar />

      {!!tab && (
        <TabInformativo
          tabs={tabs}
          currentTab={tab}
          currentStep={paso}
          valipStep={currentStep.paso}
          steps={componentsFormulario}
        />
      )}

      {!!step && <Step currentStep={step} valipStep={currentStep.valipStep} steps={steps} />}
      <Component {...servicesData} validate={currentStep.paso === paso} />
      <SvgPatronesSolicitud className="fixed-left-bottom" />
    </>
  );
};

SolicitudContainer.propTypes = {
  pageComponent: PropTypes.any.isRequired,
  servicesData: PropTypes.any.isRequired,
};

export default SolicitudContainer;
