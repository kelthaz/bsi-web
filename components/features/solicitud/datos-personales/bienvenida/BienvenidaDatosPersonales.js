import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useRouter } from 'next/router';
import ResultSimulador from '../../../../core/simulador/ResultSimulador';
import { nextStepDatosPersonales } from '../../../../../redux/actions/solicitud';
import { GRAN_SALTO_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';
import { SIMULADOR_ROUTE } from '../../../../../constants/routes/publico/publico';

const BienvenidaDatosPersonales = () => {
  const { dataSimulador, resultSimulador } = useSelector((state) => state.simulador);
  const { currentStep } = useSelector((state) => state.solicitud);
  const dispatch = useDispatch();
  const { push } = useRouter();

  const handleComienza = async () => {
    dispatch(
      nextStepDatosPersonales({
        currentStep: { ...currentStep, paso: 1 },
      })
    );
    await push(GRAN_SALTO_DATOS_PERSONA_ROUTE);
  };

  return (
    <div className="contedor-fixed">
      <div className="contedor-solicitud">
        <div className="container p-0">
          <h2 className="color-blue-storm">¡Te damos la bienvenida!</h2>
          <p className="body2 color-gray-dark">
            Nos da mucho gusto que inicies tu solicitud, no podemos esperar a escuchar qué grandes planes tienes para tu
            crédito.
          </p>
          <p className="body2 color-gray-dark">
            Antes de comenzar te confirmamos que la solicitud que estás por iniciar es por un crédito con las siguientes
            características:
          </p>
          <div className="py-3">
            <div className="card-simple-border-blue-storm">
              <ResultSimulador dataSimulador={dataSimulador} resultSimulador={resultSimulador} />
            </div>
          </div>
          <p className="body2 color-gray-dark">
            Si quieres cambiar el monto o las características de tu crédito da clic en “Cambia tu crédito”. Te
            recordamos que esta será la única ocasión en que podrás modificar el monto.
          </p>
          <div className="row">
            <div className="col-lg-6 col-md-6 col-xs-12 col-sm-12 order-lg-first order-md-first order-sm-last order-xs-last my-3">
              <div className="center-first-button">
                <button type="button" className="btn-big-inverted-secondary" onClick={() => push(SIMULADOR_ROUTE)}>
                  Cambia tu crédito
                </button>
              </div>
            </div>
            <div className="col-lg-6 col-md-6 col-xs-12 col-sm-12 order-lg-last order-md-last order-sm-first order-xs-first my-3">
              <div className="center-second-button">
                <button type="submit" className="btn-big" onClick={handleComienza}>
                  Comienza tu solicitud
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default BienvenidaDatosPersonales;
