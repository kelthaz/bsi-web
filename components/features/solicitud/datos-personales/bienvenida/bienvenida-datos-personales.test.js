import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { useRouter } from 'next/router';
import storeTest from '../../../../../redux/storeTest';
import BienvenidaDatosPersonales from './BienvenidaDatosPersonales';
import { GRAN_SALTO_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';
import { SIMULADOR_ROUTE } from '../../../../../constants/routes/publico/publico';

describe('Pruebas en el componente BienvenidaDatosPersonales', () => {
  const store = storeTest();
  const {
    simulador,
    solicitud: { currentStep },
  } = store.getState();
  simulador.dataSimulador.plazo = { label: '12 meses' };
  simulador.dataSimulador.periodicidad = { label: 'Mensuales' };
  simulador.dataSimulador.aniosEmpresa = { label: 'Menos de 2 aÃ±os' };
  simulador.dataSimulador.ventasAnio = { label: 'Menos de $2 MDP' };
  const push = jest.fn();
  useRouter.mockImplementation(() => ({ push }));

  const wrapper = mount(
    <Provider store={store}>
      <BienvenidaDatosPersonales />
    </Provider>
  );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de guardar en el reducer el nuevo paso y enviarlo al siguiente paso', async () => {
    // arrange
    const comienzaSolicitud = wrapper.find('button').at(1);
    // act
    comienzaSolicitud.simulate('click');
    // assert
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso + 1);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(GRAN_SALTO_DATOS_PERSONA_ROUTE);
  });

  test('Debe  enviarlo al simulador', async () => {
    // arrange
    const cambiaCredito = wrapper.find('button').at(0);
    // act
    cambiaCredito.simulate('click');
    // assert
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(SIMULADOR_ROUTE);
  });
});
