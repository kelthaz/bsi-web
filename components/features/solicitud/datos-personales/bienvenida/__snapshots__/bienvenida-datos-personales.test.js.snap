// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente BienvenidaDatosPersonales Debe de mostrarse correctamente el componente 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BienvenidaDatosPersonales>
    <div
      className="contedor-fixed"
    >
      <div
        className="contedor-solicitud"
      >
        <div
          className="container p-0"
        >
          <h2
            className="color-blue-storm"
          >
            ¡Te damos la bienvenida!
          </h2>
          <p
            className="body2 color-gray-dark"
          >
            Nos da mucho gusto que inicies tu solicitud, no podemos esperar a escuchar qué grandes planes tienes para tu crédito.
          </p>
          <p
            className="body2 color-gray-dark"
          >
            Antes de comenzar te confirmamos que la solicitud que estás por iniciar es por un crédito con las siguientes características:
          </p>
          <div
            className="py-3"
          >
            <div
              className="card-simple-border-blue-storm"
            >
              <ResultSimulador
                color="storm"
                dataSimulador={
                  Object {
                    "aniosEmpresa": Object {
                      "label": "Menos de 2 años",
                    },
                    "monto": 6200000,
                    "periodicidad": Object {
                      "label": "Mensuales",
                    },
                    "plazo": Object {
                      "label": "12 meses",
                    },
                    "ventasAnio": Object {
                      "label": "Menos de $2 MDP",
                    },
                  }
                }
                resultSimulador={
                  Object {
                    "cat": "",
                    "comisionApertura": "",
                    "pago": "",
                    "tasaMoratoria": "",
                    "tasaOrdinaria": "",
                  }
                }
              >
                <div
                  className="row"
                >
                  <div
                    className="text-left order-md-1 col-xs-6 col-sm-6 col-md-4 col-lg-3"
                  >
                    <h4
                      className="color-blue-storm"
                    >
                      $6,200,000.00
                    </h4>
                    <p
                      className="body3 color-gray"
                    >
                      Monto solicitado
                    </p>
                  </div>
                  <div
                    className="text-left order-md-2 col-xs-6 col-sm-6 col-md-4 col-lg-3 "
                  >
                    <h4
                      className="color-blue-storm"
                    >
                       anual
                    </h4>
                    <p
                      className="body3 color-gray"
                    >
                      Tasa ordinaria
                    </p>
                  </div>
                  <div
                    className="text-left order-md-4 col-xs-6 col-sm-6 col-md-4 col-lg-3"
                  >
                    <h4
                      className="color-blue-storm"
                    >
                      12 meses
                    </h4>
                    <p
                      className="body3 color-gray"
                    >
                      Plazo del crédito
                    </p>
                  </div>
                  <div
                    className="text-left order-md-5  order-5 col-xs-6 col-sm-6 col-md-4 col-lg-3"
                  >
                    <h4
                      className="color-blue-storm"
                    />
                    <p
                      className="body3 color-gray"
                    >
                      CAT
                    </p>
                  </div>
                  <div
                    className="text-left order-md-6 col-xs-6 col-sm-6 col-md-4 col-lg-3"
                  >
                    <h4
                      className="color-blue-storm"
                    />
                    <p
                      className="body3 color-gray"
                    >
                      Comisión por apertura
                    </p>
                  </div>
                  <div
                    className="text-left order-xs-4 order-md-7 col-xs-6 col-sm-6 col-md-4 col-lg-3"
                  >
                    <h4
                      className="color-blue-storm"
                    >
                      Mensual
                    </h4>
                    <p
                      className="body3 color-gray"
                    >
                      Esquema de pago
                    </p>
                  </div>
                  <div
                    className="text-left order-md-8 col-xs-6 col-sm-6 col-md-4 col-lg-3"
                  >
                    <h4
                      className="color-blue-storm"
                    />
                    <p
                      className="body3 color-gray"
                    >
                      Pagos 
                      mensuales
                    </p>
                  </div>
                </div>
              </ResultSimulador>
            </div>
          </div>
          <p
            className="body2 color-gray-dark"
          >
            Si quieres cambiar el monto o las características de tu crédito da clic en “Cambia tu crédito”. Te recordamos que esta será la única ocasión en que podrás modificar el monto.
          </p>
          <div
            className="row"
          >
            <div
              className="col-lg-6 col-md-6 col-xs-12 col-sm-12 order-lg-first order-md-first order-sm-last order-xs-last my-3"
            >
              <div
                className="center-first-button"
              >
                <button
                  className="btn-big-inverted-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Cambia tu crédito
                </button>
              </div>
            </div>
            <div
              className="col-lg-6 col-md-6 col-xs-12 col-sm-12 order-lg-last order-md-last order-sm-first order-xs-first my-3"
            >
              <div
                className="center-second-button"
              >
                <button
                  className="btn-big"
                  onClick={[Function]}
                  type="submit"
                >
                  Comienza tu solicitud
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </BienvenidaDatosPersonales>
</Provider>
`;
